server/channels/app/oauth.go

```diff
@@ -822,16 +822,18 @@ func (a *App) AuthorizeOAuthUser(c request.CTX, w http.ResponseWriter, r *http.R
 	stateEmail := stateProps["email"]
 	stateAction := stateProps["action"]
 	if stateAction == model.OAuthActionEmailToSSO && stateEmail == "" {
+		c.Logger().Error("No email provided in state when trying to switch from email to SSO")
 		return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest)
 	}
 
 	cookie, cookieErr := r.Cookie(CookieOAuth)
 	if cookieErr != nil {
-		return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest)
+		return nil, "", stateProps, nil, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest).Wrap(cookieErr)
```

Comment:
Does the wrapper error get sent to the client? If so, is that an issue?

Reply:
> Does the wrapper error get sent to the client? If so, is that an issue?

I reckon only `T('api.user.authorize_oauth_user.invalid_state.app_error')` is returned, but the [r.Cookie() function only returns a `http.ErrNoCookie` ](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/net/http/request.go;l=426) so nothing bad would be returned there.

Reply:
All that gets returned to the client is `api.user.authorize_oauth_user.invalid_state.app_error`. Adding the `Wrap` just gives us extra info in the logs